### ISSUE TITLE
Halftime Team Switch

### ISSUE THREAD

**Issue by SuplenC** [splewis/csgo-retakes#102]
Hi splewis! I have a hard time trying to set up the server for 2v2 tournament. I installed your retake mod and pug mod too, I did configured both pretty well so now only ".ready" is necessary to start the retake, but my problem is to set up midgame teamswitch. I set up mp_halftime 15, but after the halftime only the result switches, players stay in the same team. Would it be possible to set it up or you have to update that part?

**Comment by splewis** [comment]
You probably want to set ``sm_retakes_auto_set_teams 0``. Otherwise the plugin manages the teams and puts the previous round winners on T side.

**Comment by SuplenC** [comment]
I did set it up in retakes.cfg. I can put whole configs here if you want.

**Comment by Useless39rus** [comment]
mp_halftime 1 does not work, tried sm_retakes_auto_set_teams 0.
I also try to configure the server for a 2vs2 tournament. Correct the plugin please.

Sorry for my English.

**Comment by splewis** [comment]
https://github.com/splewis/csgo-retakes/commit/77bb2d3bc055dfbe8787a03d4084beb76f4a46e2 is an attempted fix.

You could test by downloaded a development build from https://ci.splewis.net/job/csgo-retakes/127/ and seeing if it works. (You just need to replace the addons/sourcemod/plugins/retakes.smx file)

**Comment by Useless39rus** [comment]
I use retakes-127

My configs
cfg/sourcemod/retakes/retakes.cfg
sm_retakes_auto_set_teams "0"

cfg/sourcemod/retakes/retakes_game.cfg
mp_halftime 1

After the halftime only the result switches, players stay in the same team.

Sorry for my English.

**Comment by Useless39rus** [comment]
Upper posts this I played myself during the day.

And in the evening we played with a friend and that's what happened.

We've started.
![20170603211346_1](https://cloud.githubusercontent.com/assets/6013206/26756568/f8b898ca-48a5-11e7-9419-1cac5ed0375b.jpg)

That's how the sides switch after the first half.
![20170603212048_1](https://cloud.githubusercontent.com/assets/6013206/26756579/16727e4e-48a6-11e7-8ccc-63d569c483e2.jpg)
![20170603212112_1](https://cloud.githubusercontent.com/assets/6013206/26756584/30521f68-48a6-11e7-8c5b-b1cd1f4ef28b.jpg)

Switching teams after the first half does not work.
The result of the game is switched. T side switches over the CT, and the CT side does not switch over the T and the T side of anyone. When switching from T to CT, CT continues to play for T and put the bomb.

**Comment by splewis** [comment]
Try build [128](https://ci.splewis.net/job/csgo-retakes/128/).

**Comment by Useless39rus** [comment]
Switching sides works correctly.

After switching sides, each new round changes sides then for T then for CT.

When my friend came out of the server and I stayed alone I was playing for CT every round was setting a bomb.
![20170603222305_1](https://cloud.githubusercontent.com/assets/6013206/26756901/9917c466-48ac-11e7-8bf0-3a2ab4b18c55.jpg)
![20170603222352_1](https://cloud.githubusercontent.com/assets/6013206/26756902/9917d136-48ac-11e7-93ac-d387bd0dd174.jpg)

**Comment by splewis** [comment]
I've reverted the changes since my approach didn't work (https://github.com/splewis/csgo-retakes/commit/77bb2d3bc055dfbe8787a03d4084beb76f4a46e2 and https://github.com/splewis/csgo-retakes/commit/a71fdb71d5dcf9143c5fe19de0ab3e2f57f9c86a). I'm probably not going to get around to fixing this anytime soon.

If someone else wants to take a shot: the problem seems to be detecting that a halftime swap needs to happen before the round_announce_last_round_half like my attempt - teamlogic is done in the round_prestart event, but I think round_announce_last_round_half is called after that.

**Comment by clarkeyio** [comment]
My community hosted a retakes tournament recently, so I looked into the bug. I've fixed this and created a pull request!

I also made a simple tournaments extension that adds commands like .ready and .pause, if anyone is interested in using it, keep an eye out @finalrespawn

**Comment by JesusWithPants** [comment]
@clarkeyio Hey, You pull request doesnt work. On build 128 the same behaviour happens (teamswap on every round) and on build 129 there is no teamswap at all.

**Comment by metaphased** [comment]
I'd love to have some sort of working form of this as a few of us are looking to setting up a retake ladder of some sort - I tried fixing it myself to no avail however :(

**Comment by RavageCS** [comment]
@JesusWithPants 
There is a cvar for manually setting teams, though ive never tried it.

https://github.com/splewis/csgo-retakes/blob/master/scripting/retakes.sp#L149

**Comment by clarkeyio** [comment]
@JesusWithPants my pull request hasn't been accepted yet. If you know how to compile plugins use [the fix.](https://github.com/finalrespawn/csgo-retakes/tree/halftime)

A retakes tournament extension will be released soon. It will be on the @finalrespawn organisation with simple functionality like selecting the number of players, .ready and .pause. Watch the repository for email notifications 😄

**Comment by splewis** [comment]
@RavageCS 

This entire thread is about a bug in that cvar.

**Comment by Useless39rus** [comment]
I tested one.

At first I started playing for ST. The first half went right. The second half of the game started, the first round I have the hands of the CT, and should be T. Next rounds I have the hands of T.
![20170613191956_1](https://user-images.githubusercontent.com/6013206/27096219-97bacf52-5070-11e7-9145-36349ce3cab5.jpg)

The second time I started to play for T. The first half went right. The second half of the game began, the first round I have hands T, and should be ST. The next rounds have my ST hands, but I keep playing for T and laying the bomb.
![20170613192915_1](https://user-images.githubusercontent.com/6013206/27096239-a7e2828a-5070-11e7-8da7-49f527835584.jpg)
![20170613192925_1](https://user-images.githubusercontent.com/6013206/27096237-a7d65a78-5070-11e7-8214-c4aa2c710e22.jpg)

In both cases, writes that I'm in line, but I'm in the game, not in line. After 16 rounds the game does not end. I did not check overtime, but I do not think they work either.

Forgive me for my English, I'm writing a translator through Google.

**Comment by splewis** [comment]
Without knowing specifically what version is being tested, any test like the above isn't useful.

If you're interested you can try build 131 (https://ci.splewis.net/job/csgo-retakes/131/) which should have this fixed - with the problem that player skins are likely wrong the first round after the half. I don't know if that's fixable easily.

**Comment by clarkeyio** [comment]
The bug @Useless39rus was talking about is a recently introduced bug that valve haven't fixed yet. It happens in matchmaking too.

If you want expected behaviour while playing a tournament I would suggest setting `sm_retakes_auto_set_teams "0"` in retakes.cfg and `mp_halftime 1`, `mp_match_can_clinch 1` in retakes_game.cfg.

**Comment by Useless39rus** [comment]
@clarkeyio That's how I feel.

My configs
cfg/sourcemod/retakes/retakes.cfg
sm_retakes_auto_set_teams "0"

cfg/sourcemod/retakes/retakes_game.cfg
mp_halftime 1
mp_match_can_clinch 1

**Comment by Useless39rus** [comment]
@splewis I'm very interested. I'll check build 131 in the evening after work. I will report the results. Thanks for your work.

**Comment by metaphased** [comment]
I fired build 131 a go and still getting the same issue with sides after half time - is there any way to allow a different mod, like warmod etc, to handle the team assignments? I gave warmod a go alongside it a few times and the spawn locations work fine alongside it, and (aslong as there's no knife round) it starts the game well too. Obviously only if it's easier that way, otherwise yea people are still stuck on teams for the first round of a second half, then just bounce back and forth between teams

**Comment by Useless39rus** [comment]
@splewis There are two more problems to fix, if possible. I tested the plugin one
At first I started playing for CT. The first half went right. The second half of the game started, the first round I have the hands of the CT, and should be T. Next rounds I have the hands of T.
The second time I started to play for T. The first half went right. The second half of the game began, the first round I have hands T, and should be CT. The next rounds have my CT hands, but I keep playing for T and laying the bomb.
Skins hands is not important, but for the beauty and correctness of the plugin would like to fix.

**Comment by Useless39rus** [comment]
@splewis You have done a very great job on this error. Please, do not stop. Let's finish it.

**Comment by Useless39rus** [comment]
@splewis Just tested with a friend plugin. Played 1 on 1, everything works correctly and without problems. And when I tested one, there were problems.
Many thanks for your work.

**Comment by Useless39rus** [comment]
During a pause, you can put a bomb. Please correct it.

**Comment by splewis** [comment]
That clearly sounds like a different bug, so you should make a separate issue for it.

**Comment by Useless39rus** [comment]
Done https://github.com/splewis/csgo-retakes/issues/105

**Comment by Useless39rus** [comment]
My configs

cfg/sourcemod/retakes/retakes.cfg
sm_retakes_auto_set_teams "0"
sm_retakes_maxplayers "4"

There are 6 players in the game, and there should be 4, and 2 should be in line in the spectators.

![20170715230843_1](https://user-images.githubusercontent.com/6013206/28242937-00b7da7e-69ba-11e7-9319-a8fe1a595b51.jpg)

**Comment by Useless39rus** [comment]
Also during the game, T who had the bomb, went for a CT and the bomb is gone with him. T looking for a bomb, and she's already on the CT.

**Comment by splewis** [comment]
The purpose of the ``sm_retakes_auto_set_teams 0`` is to make the plugin not do anything with team changes.

``sm_retakes_maxplayers`` is supposed to be ignored if ``sm_retakes_auto_set_teams 0`` is set.

**Comment by borzaka** [comment]
Any news with this issue/request?

I would like to enable halftime switch with fixed (no autobalance, no any auto switch, user would stay where joined) teams with these settings:
- `sm_retakes_auto_set_teams 0`
- `mp_halftime 1`
